gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,912 @@
+    ) -> None:
+        """
+        Get metrics for each Spark job.
+        """
+        iteration_metrics: Dict[str, Dict[str, Any]] = {}
+        for app_id, (app_name, tracking_url) in running_apps.items():
+            try:
+                base_url = self._get_request_url(tracking_url)
+                response = self._rest_request_to_json(base_url, SPARK_APPS_PATH, app_id, "jobs")
+                application_diff_aggregated_metrics = dict.fromkeys(SPARK_APPLICATION_DIFF_METRICS.keys(), 0)
+                application_gauge_aggregated_metrics = dict.fromkeys(SPARK_APPLICATION_GAUGE_METRICS.keys(), 0)
+                iteration_metrics[app_id] = {}
+                for job in response:
+                    iteration_metrics[app_id][job["jobId"]] = job
+                    first_time_seen_job = job["jobId"] not in self._last_iteration_app_job_metrics.get(app_id, {})
+                    # In order to keep track of an application's metrics, we want to accumulate the values across all
+                    # jobs. If the values are numActiveTasks or numActiveStages - there's no problem as they're
+                    # always up-to-date and can just be summed. In case of completed jobs, only the last 1000
+                    # (configurable - spark.ui.retainedJobs) jobs will be saved and only their metrics will be sent to
+                    # us. Older jobs will be deleted , hence we can get into a situation when an old job is deleted,
+                    # and then the accumulated metric will get lower, and the diff will be negative. In order to solve
+                    # that, we only accumulate the value of newly seen jobs or the diff from the last time the value
+                    # was seen.
+                    for metric in SPARK_APPLICATION_DIFF_METRICS.keys():
+                        if first_time_seen_job:
+                            application_diff_aggregated_metrics[metric] += int(job[metric])
+                        else:
+                            application_diff_aggregated_metrics[metric] += int(job[metric]) - int(
+                                self._last_iteration_app_job_metrics[app_id][job["jobId"]][metric]
+                            )
+
+                    for metric in SPARK_APPLICATION_GAUGE_METRICS.keys():
+                        application_gauge_aggregated_metrics[metric] += int(job[metric])
+
+                tags = [f"app_name:{str(app_name)}", f"app_id:{str(app_id)}"]
+                self._set_metrics_from_json(
+                    collected_metrics, tags, application_diff_aggregated_metrics, SPARK_APPLICATION_DIFF_METRICS
+                )
+                self._set_metrics_from_json(
+                    collected_metrics, tags, application_gauge_aggregated_metrics, SPARK_APPLICATION_GAUGE_METRICS
+                )
+
+            except Exception:
+                logger.exception("Could not gather spark jobs metrics.")
```

Comment:
```suggestion
                logger.exception("Could not gather spark jobs metrics")
```

I don't like `.` in the end of log messages